datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala

```diff
@@ -104,7 +105,8 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
     if (needAnalysis && catalog != null) {
       val modularPlan = catalog.mvSession.sessionState.rewritePlan(plan).withMVTable
       if (modularPlan.find(_.rewritten).isDefined) {
-        val compactSQL = modularPlan.asCompactSQL
+        var compactSQL = modularPlan.asCompactSQL
+        compactSQL = reWriteTheSQL(modularPlan, compactSQL)
```

Comment:
only one argument `modularPlan` is enough ? compactSQL can be derived from that inside.

Reply:
yes, because `asCompactSQL ` is not a variable in ModularPlanObject. It will go to form the sql again, so sending both